### PR TITLE
Feat: webpack template generator for webpack repos

### PR DESCRIPTION
**Pull request by info-arnav**
With reference to #3768 

The following --template flag will be introduced

```
webpack init --template webpack
```

And the template webpack will act as a replacer for webpack-defaults

I have only added the basic idea of execution, please lemme know if it is interprated correctly; then ill continue.

### CODE REVIEW

File: packages/generators/src/init-generator.ts [hunk 1]
```diff
@@ -16,13 +16,27 @@ export default class InitGenerator<
...
 
+  #parseArgs(args: string[]): void {
+    const templateIndex = args.findIndex((arg) => arg === "--template");
+    if (
+      templateIndex !== -1 &&
+      templateIndex + 1 < args.length &&
+      args[templateIndex + 1] == "webpack"
+    ) {
+      this.template = args[templateIndex + 1];
+      // now if the template is webpack, then we follow the execution of webpack-defaults ?
```
Comment (alexander-akait) [inline packages/generators/src/init-generator.ts:34]: We need to move all logic of webpack-defaults here

Comment (alexander-akait) [reply packages/generators/src/init-generator.ts:34]: i.e. the same files, configuration and etc

Comment (info-arnav) [reply packages/generators/src/init-generator.ts:34]: got it, working on it

**Review by evenstensberg — CHANGES_REQUESTED** [review]
we have a gsoc project working on CWA (Create Webpack App) and we will not accept his pr at this time. Thank you for the contribution though! Maybe we can find another issue for you to solve?

**Pull request closed without merging**